R/pkg/R/SQLContext.R

```diff
@@ -744,6 +747,9 @@ read.df.default <- function(path = NULL, source = NULL, schema = NULL, ...) {
   if (is.null(source)) {
     source <- getDefaultSqlSource()
   }
+  if (source == "csv" && is.null(options[["nullValue"]])) {
```

Comment:
Yeah this is the more conservative option - I guess thats fine for now and we can revisit this if required.